Feature/home page back end

RosenRoydev opened the pull request.

RosenRoydev committed:
AngleSharp added

RosenRoydev committed:
Added service, view Model and interface for HomePage

RosenRoydev committed:
HomeNewsService implemeted with view model and interface

RosenRoydev committed:
News added in database

RosenRoydev committed:
Home News Service added in Program.cs

RosenRoydev committed:
Home service is implemented

RosenRoydev committed:
Articles added in Db

RosenRoydev committed:
Empty spaces removed

RosenRoydev committed:
For loop added for searching crypto news in first 7 pages

RosenRoydev committed:
Empty Spaces Removed

RosenRoydev committed:
For Loop corected to start from 1

VelinArbov commented on CryptoWorld.News/CryptoWorld.News.Server/Controllers/HomeNewsController.cs:
```diff
@@ -0,0 +1,32 @@
+            if (!ModelState.IsValid) 
+            {
+                return BadRequest();
+            
```
empty row

VelinArbov commented on CryptoWorld.News/CryptoWorld.News.Core/ViewModels/Home Page/HomePageNewsModel.cs:
```diff
@@ -0,0 +1,24 @@
+
+        public string ImageUrl {  get; set; }
+
+        //06.06.2024 12:33:36
```
Remove comment

VelinArbov commented on CryptoWorld.News/CryptoWorld.News.Core/Services/News/HomeNewsService.cs:
```diff
@@ -0,0 +1,123 @@
+                        urls.Add(item.GetAttribute("href"));
+                    }
+                }
+
```
Remove empty row

RosenRoydev committed:
Changed names of Controller, Service, Interface, view model

RosenRoydev committed:
If stattemt for document == null

RosenRoydev committed:
If statement for existing article

RosenRoydev committed:
Comments removed from code

VelinArbov commented on CryptoWorld.News/CryptoWorld.News.Core/Services/News/NewsService.cs:
```diff
@@ -8,36 +8,39 @@
                 var newsUrl = document.QuerySelectorAll(".topic > a");
+
+                if (document == null) { throw new Exception(); }
+              
```
Instead of throwing an Exception for now return an empty List

VelinArbov replied:
Please review this

VelinArbov commented on CryptoWorld.News/CryptoWorld.News.Core/Services/News/NewsService.cs:
```diff
@@ -0,0 +1,128 @@
+
+        public async Task<List<PageNewsModel>> HomePageNews(int pagesCount = 7)
+        {
+
```
empty row

VelinArbov commented on CryptoWorld.News/CryptoWorld.News.Core/Services/News/NewsService.cs:
```diff
@@ -0,0 +1,128 @@
+
+            if (urls != null)
+            {
+
```
empty row

VelinArbov commented on CryptoWorld.News/CryptoWorld.News.Core/Services/News/NewsService.cs:
```diff
@@ -0,0 +1,128 @@
+            Category category = new()
+            {
+                Name = categoryName ,
+                
```
empty row

VelinArbov commented on CryptoWorld.News/CryptoWorld.News.Core/ViewModels/HomePage/PageNewsModel.cs:
```diff
@@ -0,0 +1,23 @@
+
+        }
+      
+       public string Title {  get; set; } 
```
align properties

ZdravkoBorisov commented on CryptoWorld.News/CryptoWorld.News.Core/Services/News/NewsService.cs:
```diff
@@ -0,0 +1,128 @@
+            dbContext = _dbContext;
+        }
+
+        public async Task<List<PageNewsModel>> HomePageNews(int pagesCount = 7)
```
Could you please refactor this method by adding more methods? It looks like it does a lot of things. For example, you can separate the implementation into different methods - GetNewsUrls, GetPageNewsModel, GetArticles etc...

RosenRoydev replied:
The method is splitted

ZdravkoBorisov commented on CryptoWorld.News/CryptoWorld.News.Core/Services/News/NewsService.cs:
```diff
@@ -0,0 +1,128 @@
+                List<Article> articles = new List<Article>();
+                foreach (var article in homeNews)
+                {
+                    var articleModel = new Article();
```
You can simplify the object creation. For example 
```
var articleModel = new Article
{
    Title = article.Title,
    Content = article.Content,
    ImageUrl = article.ImageUrl,
    SourceId = source.Id,
    CategoryId = category.Id
};
```

RosenRoydev replied:
The object creation is simplified

ZdravkoBorisov commented on CryptoWorld.News/CryptoWorld.News.Core/Services/News/NewsService.cs:
```diff
@@ -0,0 +1,128 @@
+                    articleModel.ImageUrl = article.ImageUrl;
+                    DateTime publicationDate;
+                    string format = "dd.MM.yyyy HH:mm:ss";
+                    bool isValidDate = DateTime.TryParseExact(article.DatePublished, format, CultureInfo.InvariantCulture
```
You can simplify this with removing the DateTime publicationDate; so the bool isValidDate could be something like this ..... " `out **DateTime** publicationDate` "

RosenRoydev replied:
Done

ZdravkoBorisov commented on CryptoWorld.News/CryptoWorld.News.Core/Services/News/NewsService.cs:
```diff
@@ -0,0 +1,128 @@
+
+            for (int i = 1; i <= pagesCount; i++)
+            {
+                var document = await context.OpenAsync($"https://money.bg/finance?page={i}");
```
Could you please add this URL to the configurations? We might want to update it and if the app is running we would need to stop it. That's why it's better to add this into a configuration (currently appsettings.json)

RosenRoydev committed:
Split HomePageNews() of  3  methods

RosenRoydev committed:
Url for money.bg is added in appsettings.js
In program cs is added class for money.bg with AddScoppedmethod and is configured by method .Configure

RosenRoydev committed:
New class for Money.bg url is added in ViewModels

RosenRoydev committed:
UrlForNews is injected in the service

RosenRoydev committed:
Empty spaces removed

RosenRoydev committed:
Merge branch 'develop' into feature/Home-page-Back-end
